### PR TITLE
Unable to listen to 'conditions' changing

### PR DESCRIPTION
**Pull request by Beaudinn**
Moved event dispatcher to proper position

### CODE REVIEW

File: src/Darryldecode/Cart/Cart.php [hunk 1]
```diff
@@ -641,11 +640,15 @@ protected function validate($item)
...
         $cart = $this->getContent();
 
         $cart->put($id, new ItemCollection($item));
 
         $this->save($cart);
+
+        $this->events->fire($this->getInstanceName().'.added', array($item, $this));
     }
 
     /**
```
Comment (darryldecode) [inline src/Darryldecode/Cart/Cart.php]: Hi @Beaudinn , does this change anything? Or any issues why you moved this dispatchers?

Comment (Beaudinn) [inline src/Darryldecode/Cart/Cart.php]: Hey, Could you tell me when and how to contribute to any project?

There were no problems reported, But I had the issue that as when a condition of an item was changed no event get triggered. In place of the triggering the update when adding an item, it seems better to do this inside the update function itself.

Comment (darryldecode) [inline src/Darryldecode/Cart/Cart.php]: @Beaudinn , so this changes is just a matter of preference right? By the way PR's are welcome..

Comment (darryldecode) [inline src/Darryldecode/Cart/Cart.php]: @Beaudinn , gotcha! I see now what you're talking about..

Comment (Beaudinn) [inline src/Darryldecode/Cart/Cart.php]: Yes it is. The only thing that has changed is the first parameter of the event, which was first only the item object but can now be also an array with changeable data.

**Pull request merged**